src/modules/categories/services/categories.service.ts

```diff
@@ -31,7 +33,7 @@ export class CategoriesServiceImpl extends CategoriesService {
 
   public override async getCategoryById(id: number): Promise<CategoriesDTO | null> {
     const category = this._categoryRepository.findOneBy({ id });
-    if (!category) { throw new BadRequestException('Can`t find category');}
+    if (!category) { throw new NotFoundException('Can`t find category');}
     return category;
```

Comment:
```suggestion
    const response = new CategoriesDTO();
    response.id = category.id;
    ...
    return response;
```

Reply:
або додати в CategoriesDTO конструктор і передавати туди ентіті

```suggestion
    return new CategoryDTO(category);
```

приклад:


export class CategoryDTO {

  public id: string;
  
  ...

  constructor(entity: CategoryEntity) {
    this.id = entity.id;
    ...
  }

}